<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PPUDebugInfoDialog</class>
 <widget class="QDialog" name="PPUDebugInfoDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>602</width>
    <height>549</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PPU Debug Info</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_7">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>PPU Control</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QCheckBox" name="checkPpuControl_AddressIncremnet">
          <property name="text">
           <string>Address Increment</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuControl_SpritePatternTable">
          <property name="text">
           <string>Sprite Pattern Table</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuControl_BackgroundPatternTable">
          <property name="text">
           <string>Background Pattern Table</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuControl_SpriteSize">
          <property name="text">
           <string>Sprite Size</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuControl_GenerateNMI">
          <property name="text">
           <string>Generate NMI</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_4">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>PPU Mask</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QCheckBox" name="checkPpuMask_Greyscale">
          <property name="text">
           <string>Greyscale</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuMask_ShowLeftmostBackground">
          <property name="text">
           <string>Show leftmost background</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuMask_ShowLeftmostSprites">
          <property name="text">
           <string>Show leftmost sprites</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuMask_ShowBackground">
          <property name="text">
           <string>Show background</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuMask_ShowSprites">
          <property name="text">
           <string>Show sprites</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuMask_IntensifyReds">
          <property name="text">
           <string>Intensify reds</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuMask_IntensifyGreens">
          <property name="text">
           <string>Intensify greens</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuMask_IntensifyBlues">
          <property name="text">
           <string>Intensify blues</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>PPU Status</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QCheckBox" name="checkPpuStatus_SpriteOverflow">
          <property name="text">
           <string>Sprite overflow</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuStatus_Sprite0Hit">
          <property name="text">
           <string>Sprite #0 Hit</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkPpuStatus_VBlankStarted">
          <property name="text">
           <string>VBlank started</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_6">
       <property name="title">
        <string>Timing</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>PPU cycle</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineTiming_PPUCycle">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Scanline</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineTiming_Scanline">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkTiming_IsEvenFrame">
          <property name="text">
           <string>Is Even Frame</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_5">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QGroupBox" name="groupBox_5">
       <property name="title">
        <string>Current PPU Address</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>Coarse X Scroll</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineCurrent_CoarseXScroll">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Coarse Y Scroll</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineCurrent_CoarseYScroll">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Nametable select</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineCurrent_NametableSelect">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Fine Y Scroll</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineCurrent_FineYScroll">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>Fine X Scroll</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineCurrent_FineXScroll">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkCurrent_FirstWrite">
          <property name="text">
           <string>First write</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="title">
        <string>Temporary PPU Address</string>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Coarse X Scroll</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineTemporary_CoarseXScroll">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Coarse Y Scroll</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineTemporary_CoarseYScroll">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Nametable select</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineTemporary_NametableSelect">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Fine Y Scroll</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineTemporary_FineYScroll">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PPUDebugInfoDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PPUDebugInfoDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
